measure

```diff
@@ -416,6 +601,121 @@ class NewRelicDriver(Measure):
                                       repr(e)))
         return result
 
+    def check_slo(self, slo_input: SloInput) -> Dict[str, List[Dict[str, Any]]]:
+        measure_to = datetime.now(timezone.utc).replace(microsecond=0)
+        measure_from = (measure_to - self.fast_fail_config.span).replace(microsecond=0)
+        
+        metrics = self.get_metrics(measure_from, measure_to)
+
+        nan_results = []
+        failed_results = []
+        passed_results = []
+        for index, condition in enumerate(slo_input.conditions):
+            # Index is used as a psuedo ID to track failures across slo_check calls
+            result = { 'condition': condition.dict(), 'condition_repr': str(condition), 'index': index }
+
+            # try to produce a scalar value from one or more metric data points, store into result['metric_scalar']
+            if not aggregate_metric_values_to_scalar(condition.metric, metrics, result, is_threshold=False):
+                nan_results.append(result)
+                continue
+
+            if isinstance(condition, ThresholdConstantCondition):
+                result['threshold_product'] = condition.threshold * condition.threshold_multiplier
+            elif isinstance(condition, ThresholdMetricCondition):
+                # try to produce a scalar, store into result['threshold_metric_scalar']
+                if not aggregate_metric_values_to_scalar(condition.threshold_metric, metrics, result, is_threshold=True):
+                    nan_results.append(result)
+                    continue
+
+                result['threshold_product'] = result['threshold_metric_scalar'] * condition.threshold_multiplier
+            else:
+                raise MeasureError(f'Unable to process slo condition with type {type(condition).__name__}', reason='invalid-type')
+
+            check_op = operator.lt if condition.keep == 'below' else operator.gt
+            if check_op(result['metric_scalar'], result['threshold_product']):
+                result['message'] = 'SLO passed'
+                passed_results.append(result)
+            else:
+                result['message'] = f'keep {condition.keep} not met'
+                failed_results.append(result)
+
+        return {
+            'from': str(measure_from),
+            'to': str(measure_to),
+            'missing_or_nan': nan_results, 
+            'failed': failed_results, 
+            'passed': passed_results
+        }
+
+
+    def slo_check_sleep(self, secs: float, slo_input: SloInput) -> None:
```

Comment:
why not check_slo_sleep()? (to match the related check_slo() function) - or do them both the other way (slo_check() and slo_check_sleep()).

Reply:
slo_check_sleep -> check_slo_sleep